benchmark/mnist_benchmark.py

```diff
@@ -592,23 +594,21 @@ def save_results(results: dict) -> None:
     print(f"Data has been saved to {file_name}")
 
 
-def get_solver_name(_solver: Callable) -> str:
+def get_solver_name(solver: Callable[[int], Solver]) -> str:
     """
     Get the name of the solver.
 
     This function extracts and returns the name of the solver class.
-    If the `_solver` is an instance of the `MapReduce` class, it retrieves the
-    name of the `base_solver` class instead.
+    If ``_solver`` is an instance of :class:`~coreax.solvers.MapReduce`, it retrieves
+    the name of the :class:`~coreax.solvers.MapReduce.base_solver` class instead.
 
-    :param _solver: An instance of a solver, such as `MapReduce` or `RandomSample`.
+    :param solver: An instance of a solver, such as `MapReduce` or `RandomSample`.
     :return: The name of the solver class.
     """
-    solver_name = (
-        _solver.base_solver.__class__.__name__
-        if _solver.__class__.__name__ == "MapReduce"
-        else _solver.__class__.__name__
-    )
-    return solver_name
+    if (rtn_type := signature(solver).return_annotation) is MapReduce:
```

Comment:
`signature(an instance of RandomSample)` gives the following error
`TypeError: RandomSample(coreset_size=25, random_key=u32[2], weighted=False, unique=True) is not a callable object`, this could be because `solver` is a `Solver` and not a `Callable`